cmd/zfs/zfs_main.c

```diff
@@ -905,6 +907,7 @@ zfs_do_create(int argc, char **argv)
 			    zfs_prop_to_name(ZFS_PROP_VOLSIZE), intval) != 0)
 				nomem();
 			volsize = intval;
+			volsize = ROUND_UP_128K(intval);
```

Comment:
You can use the existing `P2ROUNDUP(x, align)` macro here instead of adding a new one.  It allows you to round `x` up to the next `align` boundary when working with powers of two.